common/logger/src/metadata.rs

```diff
@@ -23,24 +24,34 @@ pub struct Metadata {
 
     /// The file name and line number together 'file:line'
     location: &'static str,
+
+    /// The program backtrace taken when the event occurred. Backtraces are
+    /// only supported for errors.
+    backtrace: Option<Backtrace>,
 }
 
 impl Metadata {
-    pub const fn new(
+    pub fn new(
         level: Level,
         target: &'static str,
         module_path: &'static str,
         file: &'static str,
         line: u32,
         location: &'static str,
     ) -> Self {
+        let backtrace = match level {
+            Level::Error => Some(Backtrace::new()),
+            _ => None,
+        };
```

Comment:
So, this doesn't give the backtrace of the error though, just the backtrace of the error log line.  Is this what we want?  Sometimes the error is passed several places before it's logged.

Additionally, sometimes it can be helpful for errors that were passed into warn.

Reply:
Yeah, exactly -- this takes the backtrace at the line of the "error!" log. See my response below for whether or not this makes sense right now